src/Elastic.Apm/Report/Serialization/StringTruncationValueResolver.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Elastic.Apm.Helpers;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace Elastic.Apm.Report.Serialization
+{
+	/// <inheritdoc />
+	/// <summary>
+	/// Automatically applies <see cref="StringValueProvider"/> to <see cref="String"/> properties, and <see cref="StringDictionaryValueProvider"/>
+	/// to <see cref="Dictionary{String, String}"/> unless they are marked with <see cref="NoTruncationInJsonNetAttribute"/>.
+	/// </summary>
+	internal class StringTruncationValueResolver : CamelCasePropertyNamesContractResolver
```

Comment:
Could potentially just be two `JsonConverter` e.g `TrimmedStringJsonConverter` and ` TagsJsonConverter` (for the `Dictionary<string, string>` case handled here as well. 

This would eschew the custom contract resolver and value providers.

Reply:
👍 Agreed, it makes things easier. Added the two converters, removed the contract resolver and the value provider.